PC/winreg.c

```diff
@@ -1655,6 +1674,10 @@ winreg_SaveKey_impl(PyObject *module, HKEY key, const Py_UNICODE *file_name)
     Py_RETURN_NONE;
 }
 
+#endif /* MS_WINDOWS_GAMES */
+
+#ifndef MS_WINDOWS_GAMES
```

Comment:
Here's an implementation of `winreg.SetValue()` that's based on `RegCreateKeyExW()` and `RegSetValueExW()`. 

```c
static PyObject *
winreg_SetValue_impl(PyObject *module, HKEY key, const Py_UNICODE *sub_key,
                     DWORD type, PyObject *value_obj)
/*[clinic end generated code: output=d4773dc9c372311a input=bf088494ae2d24fd]*/
{
    LONG rc;
    HKEY childKey;
    LPWSTR value;
    Py_ssize_t size;
    Py_ssize_t length;
    PyObject *result = NULL;

    if (type != REG_SZ) {
        PyErr_SetString(PyExc_TypeError, "type must be winreg.REG_SZ");
        return NULL;
    }

    value = PyUnicode_AsWideCharString(value_obj, &length);
    if (value == NULL) {
        return NULL;
    }

    size = (length + 1) * sizeof(WCHAR);
    if ((Py_ssize_t)(DWORD)size != size) {
        PyErr_SetString(PyExc_OverflowError, "value is too long");
        goto exit;
    }

    if (PySys_Audit("winreg.SetValue", "nunu#",
                    (Py_ssize_t)key, sub_key, (Py_ssize_t)type,
                    value, length) < 0)
    {
        goto exit;
    }

    if (key == HKEY_PERFORMANCE_DATA) {
        PyErr_SetFromWindowsErrWithFunction(ERROR_INVALID_HANDLE,
                                            "RegSetValue");
        goto exit;
    }

    if (!sub_key || !sub_key[0]) {
        childKey = key;
    }
    else {
        Py_BEGIN_ALLOW_THREADS
        rc = RegCreateKeyExW(key, sub_key, 0, NULL, 0, KEY_SET_VALUE, NULL,
                             &childKey, NULL);
        Py_END_ALLOW_THREADS
        if (rc != ERROR_SUCCESS) {
            PyErr_SetFromWindowsErrWithFunction(rc, "RegCreateKeyEx");
            goto exit;
        }
    }

    Py_BEGIN_ALLOW_THREADS
    rc = RegSetValueExW(childKey, NULL, 0, REG_SZ, (LPBYTE)value, (DWORD)size);
    Py_END_ALLOW_THREADS
    if (rc == ERROR_SUCCESS) {
        result = Py_NewRef(Py_None);
    }
    else {
        PyErr_SetFromWindowsErrWithFunction(rc, "RegSetValueEx");
    }

exit:
    PyMem_Free(value);
    if (childKey != key) {
        RegCloseKey(childKey);
    }
    return result;
}
```

Reply:
thanks for providing an implementation of the two functions